rpcs3/Emu/Cell/SPULLVMRecompiler.cpp

```diff
@@ -5980,39 +6007,208 @@ class spu_llvm_recompiler : public spu_recompiler_base, public cpu_translator
 			}
 		});
 
-		const auto [a, b, c] = get_vrs<f32[4]>(op.ra, op.rb, op.rc);
+		register_intrinsic("spu_re_acc", [&](llvm::CallInst* ci)
+		{
+			const auto div = value<f32[4]>(ci->getOperand(0));
+			return fsplat<f32[4]>(1.0f) / div;
+		});
 
+		const auto [a, b, c] = get_vrs<f32[4]>(op.ra, op.rb, op.rc);
 		static const auto MT = match<f32[4]>();
 
-		// Match sqrt
-		if (auto [ok_fnma, a1, b1] = match_expr(a, fnms(MT, MT, fsplat<f32[4]>(1.00000011920928955078125))); ok_fnma)
+		auto check_sqrt_pattern_for_float = [&](f32 float_value) -> bool
 		{
-			if (auto [ok_fm2, a2] = match_expr(b, fm(MT, fsplat<f32[4]>(0.5))); ok_fm2 && a2.eq(b1))
+			auto match_fnms = [&](f32 float_value)
+			{
+				auto res = match_expr(a, fnms(MT, MT, fsplat<f32[4]>(float_value)));
+				if (std::get<0>(res))
+					return res;
+
+				return match_expr(b, fnms(MT, MT, fsplat<f32[4]>(float_value)));
+			};
+
+			auto match_fm_half = [&]()
+			{
+				auto res = match_expr(a, fm(MT, fsplat<f32[4]>(0.5)));
+				if (std::get<0>(res))
+					return res;
+
+				res = match_expr(a, fm(fsplat<f32[4]>(0.5), MT));
```

Comment:
Since the order doesnt matter, this change dhould be in llvm_mul struct not here.

Reply:
But we're testing for intrinsics not for llvm_mul. And llvm_mul is not generated until the intrinsic has been evaluated.

Reply:
Okay, I will implement a mechanism for it then.